tests/tests_unit/test_api/test_agents_actions.py

```diff
@@ -0,0 +1,197 @@
+from __future__ import annotations
+
+from unittest.mock import MagicMock
+
+import pytest
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes.agents import Message
+from cognite.client.data_classes.agents.chat import (
+    ActionCall,
+    AgentChatResponse,
+    ClientToolAction,
+    ClientToolCall,
+    ClientToolResult,
+    TextContent,
+    UnknownActionCall,
+)
+
+
+@pytest.fixture
+def action_call_response_body() -> dict:
+    """Response body when agent requests an action."""
+    return {
+        "agentId": "my_agent",
+        "agentExternalId": "my_agent",
+        "response": {
+            "cursor": "cursor_12345",
+            "messages": [
+                {
+                    "content": {"text": "", "type": "text"},
+                    "actions": [
+                        {
+                            "type": "clientTool",
+                            "actionId": "call_abc123",
+                            "clientTool": {
+                                "name": "add",
+                                "arguments": '{"a": 42, "b": 58}',
+                            },
+                        }
+                    ],
+                    "role": "agent",
+                }
+            ],
+            "type": "result",
+        },
+    }
+
+
+@pytest.fixture
+def final_response_body() -> dict:
+    """Final response after action execution."""
+    return {
+        "agentId": "my_agent",
+        "agentExternalId": "my_agent",
+        "response": {
+            "cursor": "cursor_67890",
+            "messages": [
+                {
+                    "content": {"text": "The result is 100.", "type": "text"},
+                    "role": "agent",
+                }
+            ],
+            "type": "result",
+        },
+    }
+
+
+class TestClientToolCall:
+    def test_dump_serializes_arguments_to_json(self) -> None:
+        call = ClientToolCall(
+            action_id="call_456",
+            name="multiply",
+            arguments={"x": 5, "y": 7},
+        )
+
+        dumped = call.dump(camel_case=True)
+
+        assert dumped["type"] == "clientTool"
+        assert dumped["actionId"] == "call_456"
+        assert dumped["clientTool"]["name"] == "multiply"
```

Comment:
Test all with one line then you get all issues reported at once. 

```suggestion
        assert dumped == {<expected>}
```